pinot-broker/src/main/java/com/linkedin/pinot/broker/broker/BrokerAdminApiApplication.java

```diff
@@ -0,0 +1,125 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.broker.broker;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.pinot.broker.requesthandler.BrokerRequestHandler;
+import com.linkedin.pinot.broker.routing.TimeBoundaryService;
+import com.linkedin.pinot.common.metrics.BrokerMetrics;
+import io.swagger.jaxrs.config.BeanConfig;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import org.glassfish.grizzly.http.server.CLStaticHttpHandler;
+import org.glassfish.grizzly.http.server.HttpServer;
+import org.glassfish.hk2.utilities.binding.AbstractBinder;
+import org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory;
+import org.glassfish.jersey.jackson.JacksonFeature;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class BrokerAdminApiApplication extends ResourceConfig {
+  private static final Logger LOGGER = LoggerFactory.getLogger(BrokerAdminApiApplication.class);
+  public static final String RESOURCE_PACKAGE = "com.linkedin.pinot.broker.api.resources";
+
+  private URI baseUri;
+  private boolean started = false;
+  private HttpServer httpServer;
+  private BrokerServerBuilder brokerServerBuilder;
+  private BrokerMetrics brokerMetrics;
+  private BrokerRequestHandler brokerRequestHandler;
+  private TimeBoundaryService timeBoundaryService;
+
+  public BrokerAdminApiApplication(final BrokerServerBuilder brokerServerBuilder, final BrokerMetrics brokerMetrics,
+      final BrokerRequestHandler brokerRequestHandler, final TimeBoundaryService timeBoundaryService) {
+    this.brokerServerBuilder = brokerServerBuilder;
+    this.brokerMetrics = brokerMetrics;
+    this.brokerRequestHandler = brokerRequestHandler;
+    this.timeBoundaryService = timeBoundaryService;
+
+    packages(RESOURCE_PACKAGE);
+    register(new AbstractBinder() {
+      @Override
+      protected void configure() {
+        bind(brokerServerBuilder).to(BrokerServerBuilder.class);
+        bind(brokerMetrics).to(BrokerMetrics.class);
+        bind(brokerRequestHandler).to(BrokerRequestHandler.class);
+        bind(timeBoundaryService).to(TimeBoundaryService.class);
+      }
+    });
+
+    register(JacksonFeature.class);
```

Comment:
do we need this?

Reply:
I will remove this. I forgot to check this part after borrowing code from server admin api.

Reply:
yeh, probably not needed